drivers/gpio/gpio_nrfx.c

```diff
@@ -237,6 +241,10 @@ static int gpio_nrfx_pin_interrupt_configure(const struct device *port,
 					     enum gpio_int_mode mode,
 					     enum gpio_int_trig trig)
 {
+	if (!IS_ENABLED(CONFIG_GPIO_NRFX_INTERRUPT)) {
+		return -ENOTSUP;
+	}
+
```

Comment:
Wouldn't an `#ifdef CONFIG_GPIO_NRFX_INTERRUPT ... #else ... #endif` be better here?

Reply:
I prefer to use `IS_ENABLED` approach to keep the compiler checks here (code is opt out from the resulting code anyway so we still save memory)